### ISSUE TITLE
Create dataset loader for TotalDefMeme

### ISSUE THREAD

**Issue by SamuelCahyawijaya** [SEACrowd/seacrowd-datahub#355]
Dataloader name: `total_defense_meme/total_defense_meme.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?total_defense_meme

|     Dataset| total_defense_meme |
|-------------|---|
| Description | This is a large-scale multimodal and multi-attribute dataset containing memes about Singapore's Total Defence policy from different social media platforms. The type (Singaporean or generic), pillars (military, civil, economic, social, psychological, digital, others), topics and stances (against, neutral, supportive) of each meme are manually identified by annotators. |
| Subsets | - |
| Languages | eng |
| Tasks | Topic Classification, Stance Detection, Optical Character Recognition |
| License | Unknown (unknown) |
| Homepage | Image: https://drive.google.com/file/d/1oJIh4QQS3Idff2g6bZORstS5uBROjUUz/view, Annotations: https://gitlab.com/bottle_shop/meme/TotalDefMemes/-/tree/main  |
| HF URL | - |
| Paper URL | https://arxiv.org/pdf/2305.17911.pdf |

**Comment by akhdanfadh** [comment]
**My question is in the end.**

Based on the paper, they obtained a dataset of 7200 images. Then, they filtered it into 5301 memes with 2893 SG-related and 2408 non-SG. Among those SG-related memes, they annotated 2513 images. Quoting them,

> **Pillars, Topics & Stances Annotation** _....._ The annotators will first assign the memes’ defence pillars: military, civil, economic, social, psychological, digital, or others. _....._ Next, they annotate the relevant topic tags associated with the meme (i.e., nouns, pronouns, and phrases) in a free-text format. _....._ Lastly, the annotators annotate the meme’s stances towards the assigned pillars: support, against, or neutral.

So, for example, an image will have annotations like this:
```json
"Pillar_Stances": [
    {
        "img_4120.jpg": [
            [
                "Economic Defence",
                [
                    "Neutral",
                    "Neutral"
                ]
            ],
            [
                "Psychological Defence",
                [
                    "Against",
                    "Against"
                ]
            ]
        ]
    },
    ...
]
"Tags": [
    {
        "img_4120.jpg": [
            "Government",
            "HDB",
            "Gone",
            "Lease End",
            "Sad",
            "Disappear",
            "99-Years",
            "e-scooter law"
        ]
    },
    ...
]
```

Furthermore, they also provide the text in almost **all** images (7012 to be exact) with some OCR algorithm as follows:
```json
"Text": [
    {
        "img_4120.jpg": "When a HDB flat finishes it's 99-year lease: This is so sad can we hit a pedestrian with escooters?"
    },
    {
        "img_1712.jpg": "News; The mystery Chinese Virus can only spread through human interaction Engineering Students:"
    },
    ...
]
```

I think this one will use general [image_text](https://github.com/SEACrowd/seacrowd-datahub/blob/7bdfb4b461d6449b8200950938b13ef7614bc4f6/seacrowd/utils/schemas/image_text.py) seacrowd schema. My question is **should I just implement the text OCR set**? If so, then I can pass `tags` for `metadata/context`. But **I am not sure how to proceed with the stances labelling**.

**Comment by akhdanfadh** [comment]
#self-assign

**Comment by holylovenia** [comment]
> I think this one will use general [image_text](https://github.com/SEACrowd/seacrowd-datahub/blob/7bdfb4b461d6449b8200950938b13ef7614bc4f6/seacrowd/utils/schemas/image_text.py) seacrowd schema. My question is **should I just implement the `text` OCR field and ignore the `pillars`**? If so, then I can pass `tags` for `metadata/context` in the schema. If not, **I am not sure how to proceed with the stance labeling**.

I agree with you.
- The OCR subsets will use the `[image_text](https://github.com/SEACrowd/seacrowd-datahub/blob/7bdfb4b461d6449b8200950938b13ef7614bc4f6/seacrowd/utils/schemas/image_text.py)` schema. The `pillars` and `tags` info can simply be stored inside `meta`.
- The topic classification subsets and the vision-language stance labeling subsets could use a new schema for image classification. We don't have one right now. Could you please make a separate PR for this new schema? According to our running name convention, the schema probably should be named `image` (though it sounds kind of weird...)

I hope this clears up things.

**Comment by akhdanfadh** [comment]
> The OCR subsets will use the [image_text](https://github.com/SEACrowd/seacrowd-datahub/blob/7bdfb4b461d6449b8200950938b13ef7614bc4f6/seacrowd/utils/schemas/image_text.py) schema. The `pillars` and `tags` info can simply be stored inside `meta`.

The `metadata` on the schema is organized like this. The `tags` can be passed into `context` but I'm not sure about the `pillars`. Is it okay if I add another key in the metadata schema, for example, by using this code: `feature['metadata']['stances'] = ...`? See an implementation [here](https://github.com/SEACrowd/seacrowd-datahub/blob/32620b1d752b399cbdb591155133b7366fa48ace/seacrowd/sea_datasets/tatabahasa/tatabahasa.py#L101) for reference.
```python
"metadata": {
    "context": datasets.Value("string"),
    "labels": datasets.Sequence(datasets.ClassLabel(names=label_names)),
}
```
---
> The topic classification subsets and the vision-language stance labeling subsets could use a new schema for image classification. We don't have one right now. Could you please make a separate PR for this new schema? According to our running name convention, the schema probably should be named `image` (though it sounds kind of weird...)

Hmm, I don't mind actually. Though, implementing image classification schema would mean this SEACrowd project is not entirely NLP-oriented anymore haha 😄. I'll create a PR by this weekend.

**Comment by holylovenia** [comment]
> Hmm, I don't mind actually. Though, implementing image classification schema would mean this SEACrowd project is not entirely NLP-oriented anymore haha 😄. I'll create a PR by this weekend.

Yes definitely. 👍 We're also consolidating every VL and speech and other datasets we can get our hands on.
